src/encoded/types/file.py

```diff
@@ -1191,14 +1229,143 @@ def download(context, request):
+        # Product "Variant" (supposed to be like black, gray, etc), we repurpose for filetype for reporting
+        "pr1va": properties.get("file_type", "other") # "other" MATCHES THAT IN `file_type_detaild` calc property, since file_type_detailed is used on frontend when performing "Select All" files.
+    }
+
+    # Custom dimensions
+    # See https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#pr_cm_
+    if "name" in ga_dimension_map:
+        ga_payload["dimension" + ga_dimension_map["name"]] = filename
+        ga_payload["pr1cd" + ga_dimension_map["name"]] = filename
+
+    if "experimentType" in ga_dimension_map:
+        ga_payload["dimension" + ga_dimension_map["experimentType"]] = file_experiment_type or None
+        ga_payload["pr1cd" + ga_dimension_map["experimentType"]] = file_experiment_type or None
+
+    if "filesize" in ga_metric_map:
+        ga_payload["metric" + ga_metric_map["filesize"]] = file_size_downloaded
+        ga_payload["pr1cm" + ga_metric_map["filesize"]] = file_size_downloaded
+
+    if "downloads" in ga_metric_map:
+        ga_payload["metric" + ga_metric_map["downloads"]] = 0 if request.range else 1
+        ga_payload["pr1cm" + ga_metric_map["downloads"]] = 0 if request.range else 1
+
+
+    # client id (`cid`) or user id (`uid`) is required. uid shall be user uuid.
+    # client id might be gotten from Google Analytics cookie, but not stable to use and wont work on programmatic requests...
+    if user_uuid:
+        ga_payload['uid'] = user_uuid
+    if ga_cid:
+        ga_payload['cid'] = ga_cid
+
+    # ga_payload_event = dict({
+    #     "t": "event",         # Hit type. Could also be event, transaction, pageview, etc.
+    #     "ec": "File Download",      # Event Category
+    #     "e1": filename,             # Event Label
+    #     "ev": properties.get('file_size', 0),             # Event Value
+    # }, **ga_payload)
+
+    # ga_transaction = dict({
+    #     "t": "transaction",         # Hit type. Could also be event, transaction, pageview, etc.
+    #     # Product fields
+    #     "pa": "purchase",
+    #     "pr1id": context.jsonld_id(request),
+    #     "pr1nm": filename,
+    #     # TODO: "pr1br" : lab name or display_title,
+    #     "pr1qt": 1,
+    #     "pr1ca": "/".join([ ty for ty in reversed(context.jsonld_type()[:-1]) ]), # File/FileProcessed
+    #     "pr1va": file_format_name, # "Variant" (supposed to be like black, gray, etc), we repurpose for filetype for reporting
+    #     "pr1cm1": properties.get('file_size', 0), # See ga_metric_map, see https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#pr_cm_
+    #     "pr1cd2": filename,          # See ga_dimension_map
+    #     "pr1cd5": get_file_experiment_type(request, context, properties) # See ga_dimension_map
+    # } **ga_payload)
+
+
+    # TODO: WE SHOULD WRAP IN TRY/EXCEPT BLOCK RE: NETWORK?
+
+    # print('\n\n', ga_payload)
+
+    resp = requests.post(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'resp' is assigned to but never used](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=4885443)